pokemon-league-participation-manager/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/entities/Badge.java

```diff
@@ -0,0 +1,79 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.entities;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+import java.util.Objects;
+
+/**
+ * Entity class for badge object.
+ *
+ * @author Michal Mokros 456442
+ */
+@Entity(name = "Badge")
+@Table(name = "TABLE")
+public class Badge {
+
+    @Id
+    @GeneratedValue(strategy =  GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    @NotNull
+    private Trainer trainer;
+
+    @OneToMany
+    @NotNull
+    private Gym gym;
+
+    @NotNull
+    @Temporal(TemporalType.DATE)
+    private LocalDate date;
```

Comment:
We'll need a field that tells us the status of the challenge - (enum - WAITING_TO_ACCEPT, WON, LOST?).

Reply:
This still needs addressing. I suggest we discuss it in person tomorrow.

Reply:
Should this field be also part of the constructor? Or should it have some default value?

Reply:
Default value - null or WAITING_TO_ACCEPT - null probably better, since it forces the user to set it explicitly (should annotate as NotNull), and it shouldn't be included in the constructor (only id should be included).